torch/csrc/jit/script/compiler.cpp

```diff
@@ -729,7 +729,17 @@ struct to_ir {
         const auto type = classType->type_;
         Value* new_input =
             block->addInput()->setUniqueName(name)->setType(type);
-        environment_stack->setVar((*it).ident().range(), name, new_input);
+
```

Comment:
I am confused. `self` starts out as a ClassValue, then there is a check, and it either becomes a InitializingClassValue (which is not a subclass of ClassValue), or it becomes a SimpleValue. I would expect self to start out as a SimpleValue, and for ClassValue to not exist at all. I think some of this confusion is coming from the previous patch leaving the punning in between class constructors and instances of the class. I do not know which is intended by the classes in this PR.

Reply:
I don't think we can get rid of ClassValue entirely, since the class types themselves are not first-class (e.g. we don't have a `type` type). But `self` should definitely not be a `ClassValue`, will change